tkg-extensions-mods-examples/registry/harbor/harbor-extension.yaml

```diff
@@ -0,0 +1,46 @@
+apiVersion: clusters.tmc.cloud.vmware.com/v1alpha1
+kind: Extension
+metadata:
+  name: harbor
+  namespace: tanzu-system-registry
+  annotations:
+    tmc.cloud.vmware.com/managed: "false"
+spec:
+  description: harbor
+  version: "v2.0.2_vmware.1"
+  name: harbor
+  namespace: tanzu-system-registry
+  deploymentStrategy:
+    type: KUBERNETES_NATIVE
+  objects: |
+    apiVersion: kappctrl.k14s.io/v1alpha1
+    kind: App
+    metadata:
+      name: harbor
+      annotations:
+        tmc.cloud.vmware.com/orphan-resource: "true"
+    spec:
+      syncPeriod: 5m
+      serviceAccountName: harbor-extension-sa
+      fetch:
+        - image:
+            url: registry.tkg.vmware.run/tkg-extensions-templates:v1.2.0_vmware.1
+        - inline:
+            pathsFrom:
+              - configMapRef:
+                  name: harbor-overlay
+      template:
+        - ytt:
+            ignoreUnknownComments: true
+            paths:
+              - 0/tkg-extensions/common
+              - 0/tkg-extensions/registry/harbor
+              - 1/overlay-s3-pvc-fix.yaml
```

Comment:
I'm not fully understanding what this `overlay-s3-pvc-fix` does and why we need it.

Reply:
@jaimegag The out of the box Harbor extension didn't run for me when I was using S3. The pod for deployment `harbor-registry` was failing with a container creation error. I figured out it was because it was looking for a PVC for the volume `registry-data`. That PVC isn't needed when using S3.

The overlays in the shipped extension properly remove the PVC from the manifest, but theydidn't get rid of the reference to it in the deployment `harbor-registry`. The overlay I added takes care of that, changing the spec for the volume `registry-data` to `emptyDir: {}` from the PVC reference when the PVC isn't created.